middleware-jackson/src/main/java/com/nytimes/android/external/store3/middleware/jackson/JacksonReaderParser.kt

```diff
@@ -0,0 +1,33 @@
+package com.nytimes.android.external.store3.middleware.jackson
+
+
+import com.fasterxml.jackson.core.JsonFactory
+import com.fasterxml.jackson.databind.JavaType
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.KotlinModule
+import com.nytimes.android.external.store3.base.Parser
+import com.nytimes.android.external.store3.util.ParserException
+import io.reactivex.annotations.NonNull
+import java.io.Reader
+import java.lang.reflect.Type
+import javax.inject.Inject
+
+class JacksonReaderParser<Parsed> : Parser<Reader, Parsed> {
+
+    private val objectMapper: ObjectMapper
+    private val parsedType: JavaType
+
+    constructor(jsonFactory: JsonFactory, type: Type) {
+        objectMapper = ObjectMapper(jsonFactory).registerModule(KotlinModule())
```

Comment:
Maybe add `this.` here as well, to match the `@Inject` annotated constructor as well?